src/containers/treePage/index.tsx

```diff
@@ -149,62 +182,99 @@ const TreePage: React.FC<SiteProps> = ({ address, entries }) => {
         />
       </Helmet>
       <PageLayout>
-        <TreePageContainer>
-          {/*Change to tree map route once that page is finished*/}
-          <ReturnButton to={Routes.HOME}>{`<`} Return to Tree Map</ReturnButton>
-          <TreeMainContainer>
+        {asyncRequestIsComplete(siteData) && (
+          <TreePageContainer>
+            {/*Change to tree map route once that page is finished*/}
+            <ReturnButton to={Routes.HOME}>
+              {`<`} Return to Tree Map
+            </ReturnButton>
+            <TreeMainContainer>
+              <Row>
+                <Col span={17}>
+                  <TreeInfoContainer>
+                    {siteData.result.entries[0].commonName && (
+                      <PageHeader
+                        pageTitle={siteData.result.entries[0].commonName}
+                      />
+                    )}
+                    <Title level={2}>{siteData.result.address}</Title>
+                    {privilegeLevel !== PrivilegeLevel.NONE ? (
+                      <>
+                        {doesUserOwnTree ? (
+                          <Button
+                            type="primary"
+                            size="large"
+                            onClick={onClickUnadopt}
+                          >
+                            Unadopt
+                          </Button>
+                        ) : (
+                          <Button
+                            type="primary"
+                            size="large"
+                            onClick={onClickAdopt}
+                          >
+                            Adopt
+                          </Button>
+                        )}
+
+                        <StewardshipContainer>
+                          <Title level={3}>Record Tree Care</Title>
+                          <StewardshipForm
+                            onFinish={onFinishRecordStewardship}
+                            form={stewardshipFormInstance}
+                          />
+                        </StewardshipContainer>
+                      </>
+                    ) : (
+                      <Paragraph>Log in to adopt this tree!</Paragraph>
```

Comment:
⛏️ Maybe a link to login? Would also allow redirect back to this site after logging in

Reply:
Not super familiar with router and redirecting, right now the user is redirected to the home page. Do you know how to redirect the user back to this page after login?

Reply:
I think you would use something like `history.push(Routes.LOGIN, { destination: currentLocation })`. If you push a location with a state object with `RedirectStateProps`, login can get the desired destination.